src/Core/Domain/Product/SpecificPrice/ValueObject/FixedPrice.php

```diff
@@ -29,16 +29,32 @@
 
 use PrestaShop\Decimal\DecimalNumber;
 
-class FixedPrice extends DecimalNumber
+class FixedPrice implements FixedPriceInterface
 {
     /**
-     * Inherited from legacy.
-     * SpecificPrice->price can have certain value which means that product initial price should be used.
+     * @var DecimalNumber
      */
-    public const LEAVE_PRODUCT_INITIAL_PRICE_VALUE = '-1';
+    private $value;
 
-    public function isInitialPrice(): bool
+    public function __construct(
+        string $value
+    ) {
+        $this->setValue($value);
+    }
+
+    public function getValue(): DecimalNumber
     {
-        return $this->equals(new FixedPrice(self::LEAVE_PRODUCT_INITIAL_PRICE_VALUE));
+        return $this->value;
+    }
+
+    private function setValue(string $value): void
```

Comment:
I'm not sure a setter function is necessary, rather a validation function to assert the input value